lab_2/main.py

```diff
@@ -0,0 +1,133 @@
+import requests
+import csv
+from datetime import datetime, timedelta
+from dateutil.relativedelta import relativedelta
+from collections import Counter
+
+
+class MovieData:
+    api_path = 'https://api.themoviedb.org/3'
+    genre_url = f'{api_path}/genre/movie/list?language=en'
+    data_url = api_path + '/discover/movie?include_adult=false&include_video=false&sort_by=popularity.desc&page={!!!!}'
+    headers = {
+        'accept': 'application/json',
+        'Authorization': 'Bearer eyJhbGciOiJIUzI1NiJ9.eyJhdWQiOiIzMTI3NGFmYTRlNTUyMjRjYzRlN2Q0NmNlMTNkOTZjOSIsInN1YiI6IjVkNmZhMWZmNzdjMDFmMDAxMDU5NzQ4OSIsInNjb3BlcyI6WyJhcGlfcmVhZCJdLCJ2ZXJzaW9uIjoxfQ.lbpgyXlOXwrbY0mUmP-zQpNAMCw_h-oaudAJB6Cn5c8'
+    }
+
+    def __init__(self, numb_of_pages):
+        self.genres = None
+        self.pages = numb_of_pages
+        self.data = []
+
+    def fetch_all_data_in_pages(self):
+        for i in range(self.pages):
+            response = requests.get(self.data_url, params={'page': i + 1}, headers=self.headers)
+            self.data.extend(response.json()['results'])
+
+    def give_data(self):
+        return self.data
+
+    def give_data_with_indexes(self):
+        return self.data[3:19:4]
+
+    def find_popular_title(self):
+        max_popularity = max(self.data, key=lambda x: x['popularity'])
+        popular_title = max_popularity['original_title']
+        return popular_title
+
+    def find_film_with_keyword(self, keyword):
+        title_list = [j['original_title'] for j in self.data if keyword in j['overview']]
+        return title_list
+
+    def get_unique_collections(self):
+        response = requests.get(self.genre_url, headers=self.headers)
+        self.genres = response.json()['genres']
+        genre_dict = {i['id']: i['name'] for i in self.genres}
```

Comment:
I'd suggest you to create a separate function for this. and make **genre_dict** as a field